crates/egui/src/widgets/text_edit/builder.rs

```diff
@@ -481,7 +481,15 @@ impl<'t> TextEdit<'t> {
         const MIN_WIDTH: f32 = 24.0; // Never make a [`TextEdit`] more narrow than this.
         let available_width = (ui.available_width() - margin.sum().x).at_least(MIN_WIDTH);
         let desired_width = desired_width.unwrap_or_else(|| ui.spacing().text_edit_width);
-        let wrap_width = if ui.layout().horizontal_justify() {
+
+        let wrap_mode = ui.style().wrap_mode.unwrap_or(if multiline {
```

Comment:
Should we really use `TextWrapMode::Truncate` if set? I don't think that will work very well

Reply:
> TextWrapMode::Truncate

I made some changes, but I'm not sure if the changes will work everywhere.

I haven't reviewed the `TextWrapMode::Truncate` issue.
I'm not sure how `TextWrapMode::Truncate` is supposed to work with `singleline()` and `multiline()`, and what it's intended to do.
How about handling the `TextWrapMode::Truncate` issue in a new commit?

You can drop this and create a new one.